vmdb/app/models/miq_worker.rb

```diff
@@ -88,7 +88,7 @@ def self.server_scope(server_id = nil)
       server = MiqServer.my_server
       server_id = server.id unless server.nil?
     end
-    where(:miq_server_id => server_id)
+    where(:miq_server_id => server_id, :type => self.to_s)
```

Comment:
The `:type` that it's adding allows for all subclasses.  This is forcing it to stick to only the current class.